Upgrade to rails 5.0.2

Davidslv opened the pull request:
I'll rebase and clean up the commits after people had the chance to review.

Trello: https://trello.com/c/GYeuyMtX/69-upgrade-collections-to-rails-5

carvil commented on Gemfile:
```diff
@@ -1,13 +1,13 @@
 source 'https://rubygems.org'
 
-gem 'rails', '4.2.7.1'
+gem 'rails', '5.0.0'
```
We could use 5.0.2: https://rubygems.org/gems/rails

carvil commented on config/cable.yml:
```diff
@@ -0,0 +1,9 @@
+
+production:
+  adapter: redis
+  url: redis://localhost:6379/1
```
Do we need to setup action cable? I'm not sure we use it...

carvil commented on config/environments/production.rb:
```diff
@@ -29,59 +25,44 @@
-  config.assets.version = '1.0'
+  # `config.assets.precompile` and `config.assets.version` have moved to config/initializers/assets.rb
 
-  if ENV['RAILS_SERVE_STATIC_ASSETS'] == 'yes'
```
This seems to have been deleted, was it on purpose?

Davidslv replied:
It seems the option no longe exists under that name:

> config.public_file_server.enabled configures Rails to serve static files from the public directory. This option defaults to true, but in the production environment it is set to false because the server software (e.g. NGINX or Apache) used to run the application should serve static files instead. If you are running or testing your app in production mode using WEBrick (it is not recommended to use WEBrick in production) set the option to true. Otherwise, you won't be able to use page caching and request for files that exist under the public directory.

http://guides.rubyonrails.org/v5.0.0/configuring.html#configuring-assets

carvil commented on config/environments/test.rb:
```diff
@@ -26,8 +28,6 @@
-  # Print deprecation notices to the stderr.
-  config.active_support.deprecation = :stderr
-
-  config.active_support.test_order = :random
```
Was this deleted on purpose?

Davidslv replied:
The default is now `:random`

> config.active_support.test_order sets the order that test cases are executed. Possible values are :random and :sorted. This option is set to :random in config/environments/test.rb in newly-generated applications. If you have an application that does not specify a test_order, it will default to :sorted, until **Rails 5.0, when the default will become :random.**

But I'll add it back

carvil commented on config/environments/test.rb:
```diff
@@ -26,8 +28,6 @@
   config.action_controller.allow_forgery_protection = false
 
-  # Print deprecation notices to the stderr.
-  config.active_support.deprecation = :stderr
```
Same question as below.

carvil commented on config/puma.rb:
```diff
@@ -0,0 +1,47 @@
+# Puma can serve each request in a thread from an internal thread pool.
```
I think we can delete this file. Puma isn't in the Gemfile.

carvil replied:
More details: https://github.com/alphagov/govuk-developer-docs/blob/master/source/manual/upgrade-rails.html.md

Davidslv commented:
@carvil I believe I've addressed the comments.
I forgot to refer that airbrake upgrade will not work with our current errbit - although that was stated in here https://github.com/alphagov/govuk-developer-docs/blob/master/source/manual/upgrade-rails.html.md#airbrake
